[DOCS] Updates the 8.0.0 upgrade docs (#125128)
elastic/kibana: docs/setup/upgrade/upgrade-migrations.asciidoc

[[saved-object-migrations]]
=== Saved object migrations

Each time you upgrade {kib}, an upgrade migration is performed to ensure that all <<managing-saved-objects,saved objects>> are compatible with the new version.

NOTE: To help you prepare for the upgrade to 7.0.0, 6.7.0 includes an https://www.elastic.co/guide/en/kibana/6.7/upgrade-assistant.html[*Upgrade Assistant*]. 
To access the assistant, go to *Management > 7.0 Upgrade Assistant*.

WARNING: {kib} 7.12.0 and later uses a new migration process and index naming scheme. Before you upgrade, read the documentation for your version of {kib}.

> WARNING: The following instructions assumes {kib} is using the default index names. If the `kibana.index` or `xpack.tasks.index` configuration settings are different from the default, adapt the instructions accordingly.

Reviewer: The wording is "The following instructions" but there are no instructions.  Not sure how to handle this.

. Restart {kib}.
+
The dashboard with the `e3c5fc71-ac71-4805-bcab-2bcc9cc93275` ID that belongs to the `marketing_space` space **is no longer available**.

> Be sure you have a snapshot before you delete the corrupt document. If you are unable to restore from a snapshot, it is recommended to also delete the `temp` and `target` indices the migration creates before you restart {kib} and retry the snapshot restore.

Reviewer: Should this be a warning that comes before the steps?

[float]

> ===== User defined index templates that causes new `.kibana*` indices to have incompatible settings or mappings

Reviewer: ```suggestion
===== User defined index templates that cause new `.kibana*` indices to have incompatible settings or mappings
```

If {kib} unexpectedly terminates while migrating a saved object index, {kib} automatically attempts to
perform the migration again when the process restarts. Do not delete any saved objects indices to
to fix a failed migration. Unlike previous versions, {kib} 7.12.0 and
later does not require deleting indices to release a failed migration lock.

If upgrade migrations fail repeatedly, refer to
<<preventing-migration-failures, preventing migration failures>>.
When you address the root cause for the migration failure,
{kib} automatically retries the migration.

> If you're unable to resolve a failed migration, contact Support.

Reviewer: Can support be a link?

[float]
[[upgrade-migrations-rolling-back]]
==== Rolling back to a previous version of {kib}

If you've followed <<preventing-migration-failures, preventing migration failures>>
and <<resolve-migrations-failures, resolving migration failures>>, and
{kib} is still unable to successfully upgrade, rollback {kib} until
you're able to identify and fix the root cause.

> WARNING: Before you roll back {kib}, ensure that the version you want to rollback to is compatible with

Reviewer: ```suggestion
WARNING: Before you roll back {kib}, ensure that the version you want to roll back to is compatible with
```

your {es} cluster. If the version you want to roll back to is not compatible, you must also rollback {es}.
Any changes made after an upgrade are lost when you roll back to a previous version.

> To rollback after a failed upgrade migration, you must also rollback the saved object indices to be compatible with the previous {kib} version.

Reviewer: ```suggestion
To roll back after a failed upgrade migration, you must also rollback the saved object indices to be compatible with the previous {kib} version.
```

[float]

> ===== Rollback by restoring a backup snapshot

Reviewer: ```suggestion
===== Roll back by restoring a backup snapshot
```

. Before proceeding, {ref}/snapshots-take-snapshot.html[take a snapshot] that contains the `kibana` feature state.
   By default, snapshots include the `kibana` feature state.
. To make sure no {kib} instances are performing an upgrade migration, shut down all {kib} instances.

> . To delete all saved object indices, use `DELETE /.kibana*`.

Reviewer: In this and the following section of instructions, should we call out the DELETE, GET, PUT instructions in a code block as done previously in the doc?

. {ref}/snapshots-restore-snapshot.html[Restore] the `kibana` feature state from the snapshot.
. Start all {kib} instances on the older version you want to rollback to.

[float]

> ===== (Not recommended) Rollback without a backup snapshot

Reviewer: ```suggestion
===== (Not recommended) Roll back without a backup snapshot
```

. To make sure no {kib} instances are performing an upgrade migration, shut down all {kib} instances.
. {ref}/snapshots-take-snapshot.html[Take a snapshot] that includes the `kibana` feature state. By default, snapshots include the `kibana` feature state.
. Delete the version-specific indices created by the failed upgrade migration.
+
For example, to rollback from a failed upgrade
to v7.12.0, use `DELETE /.kibana_7.12.0_*,.kibana_task_manager_7.12.0_*`.
. Inspect the output of `GET /_cat/aliases`.
+
If the `.kibana` or `.kibana_task_manager` aliases are missing, you must create them manually.
Find the latest index from the output of `GET /_cat/indices` and create the missing alias to point to the latest index.
For example, if the `.kibana` alias is missing, and the latest index is `.kibana_3`, create a new alias using `POST /.kibana_3/_aliases/.kibana`.

> . To remove the write block from the rollback indices, use

Reviewer: ```suggestion
. To remove the write block from the roll back indices, use
```
